quic/s2n-quic-tests/src/tests/tls_connection_with_server_address.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::*;
+
+#[cfg(unix)]
+#[test]
+fn tls_connection_has_server_local_address_test() {
```

Comment:
Instead of testing this using events (and defining the new event, which doesn't seem particularly useful outside of this test), could we have the test operate more like how we expect a customer to use this feature? Basically define a ClientHelloCallback that is looking for the address information and assert that it is there